Shared/Samples/Authenticate with token/AuthenticateWithTokenView.swift

```diff
@@ -0,0 +1,144 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import ArcGISToolkit
+import SwiftUI
+
+struct AuthenticateWithTokenView: View {
+    /// The authenticator to handle authentication challenges.
+    @StateObject private var authenticator = Authenticator()
+    
+    /// The loaded map result.
+    @State private var mapLoadResult: Result<Map, Error>?
+    
+    /// Creates the map for this sample.
+    static func makeMap() -> Map {
+        // The portal to authenticate with named user.
+        let portal = Portal(url: .portal, connection: .authenticated)
+        
+        // The portal item to be displayed on the map.
+        let portalItem = PortalItem(
+            portal: portal,
+            id: .trafficMap
+        )
+        
+        // Creates map with portal item.
+        return Map(item: portalItem)
+    }
+    
+    var body: some View {
+        Group {
+            if let mapLoadResult = mapLoadResult {
+                switch mapLoadResult {
+                case .success(let value):
+                    MapView(map: value)
+                case .failure(let error):
+                    Text("Error loading map: \(errorString(for: error))")
+                        .padding()
+                }
+            } else {
+                ProgressView()
+                    .task {
+                        mapLoadResult = await Result { @MainActor in
+                            let map = Self.makeMap()
+                            try await map.load()
+                            try await map.operationalLayers.first?.load()
+                            return map
+                        }
+                    }
+            }
+        }
+        .authenticator(authenticator)
+        .onAppear {
+            setupAuthenticator()
+        }
+        .onDisappear {
+            Task {
+                // Reset the challenge handlers and clear credentials
+                // when the view disappears so that user is prompted to enter
+                // credentials every time the sample is run, and to clean
+                // the environment for other samples.
+                await teardownAuthenticator()
+            }
+        }
+    }
+}
+
+private extension AuthenticateWithTokenView {
+    /// The string describing an error.
+    /// - Parameter error: The error.
+    func errorString(for error: Error) -> String {
+        return if error is ArcGISChallengeCancellationError {
+            "User cancelled error"
```

Comment:
Where and how this is displayed, this string might make more sense:

```suggestion
            "User cancelled."
```